tests/UnitTests/BankTransactionUnitTest.php

```diff
@@ -30,4 +34,131 @@ public function testCanGetReferenceFromLine()
+    {
+        $line = new Total();
+        $line->setDescription($description);
+        $line->setValue(Money::EUR(0));
+
+        $bankTransaction->addLine($line);
+
+        return $line;
+    }
+
+    private function addDetailLineWithDescription(BankTransaction $bankTransaction, string $description): Detail
+    {
+        $line = new Detail();
+        $line->setDescription($description);
+        $line->setValue(Money::EUR(0));
+
+        $bankTransaction->addLine($line);
+
+        return $line;
+    }
+
+    public function testGetLinesReturnsLinesInTheSameOrderAsTheyWereAdded()
+    {
+        $total_line_count = 300;
+
+        $bank_transaction = new BankTransaction();
+
+        $this->addTotalLineWithDescription($bank_transaction, '0');
+
+        for ($i = 1; $i < $total_line_count; $i++) {
+            $this->addDetailLineWithDescription($bank_transaction, (string)$i);
+        }
+
+        self::assertLinesInOrderBasedOnDescription($bank_transaction, $total_line_count);
+    }
+
+    public function testGetLinesReturnsEmptyArrayWhenNoLinesSet()
+    {
+        $bank_transaction = new BankTransaction();
+
+        self::assertCount(0, $bank_transaction->getLines());
+    }
+
+    public function testGetLinesReturnsLinesInSameOrderWhenOnlyDetailsAdded()
+    {
+        $bank_transaction = new BankTransaction();
+
+        $this->addDetailLineWithDescription($bank_transaction, '0');
+        $this->addDetailLineWithDescription($bank_transaction, '1');
+        $this->addDetailLineWithDescription($bank_transaction, '2');
+
+        self::assertLinesInOrderBasedOnDescription($bank_transaction, 3);
+    }
+
+    public function testGetLinesReturnsLinesInSameOrderWhenOnlyTotalLineIsAdded()
+    {
+        $bank_transaction = new BankTransaction();
+
+        $this->addTotalLineWithDescription($bank_transaction, '0');
+
+        self::assertLinesInOrderBasedOnDescription($bank_transaction, 1);
+    }
+
+    public function testAdding1TotalLineAnd500DetailLinesToABankTransactionWorks()
+    {
+        $total_line_count = 501;
+
+        $bank_transaction = new BankTransaction();
+
+        $this->addTotalLineWithDescription($bank_transaction, '0');
+        for ($i = 1; $i < $total_line_count; $i++) {
+            $this->addDetailLineWithDescription($bank_transaction, (string)$i);
+        }
+
+        self::assertCount(501, $bank_transaction->getLines());
+    }
+
+    public function testAdding1TotalLineWithMoreThan500DetailLinesToABankTransactionThrows()
+    {
+        $total_line_count = 502;
+
+        $bank_transaction = new BankTransaction();
+
+        $this->addTotalLineWithDescription($bank_transaction, '0');
+
+        $this->expectException(\InvalidArgumentException::class);
+
+        for ($i = 1; $i < $total_line_count; $i++) {
+            $this->addDetailLineWithDescription($bank_transaction, (string)$i);
+        }
+    }
```

Comment:
Can you also add some tests to ensure that VAT lines are at the end of the list?

Reply:
I will add a similar test to the `SalesTransaction` so that the entire `LineField` trait is covered.

Reply:
✓